source/task_scheduler.cpp

```diff
@@ -752,9 +776,17 @@ void TaskScheduler::WaitForCounterInternal(BaseCounter *counter, unsigned value,
 	tls.OldFiberDestination = FiberDestination::ToWaiting;
 	tls.OldFiberStoredFlag = &readyFiberBundle->FiberIsSwitched;
 
+	if (m_callbacks) {
+		m_callbacks->OnFiberStateChanged(tls.OldFiberIndex, ISchedulerUserCallbacks::FiberState::DETACHED);
```

Comment:
We can replace `tls.OldFiberIndex` with `currentFiberIndex` to save a dereference.